pkg/lint/rules/chartfile.go

```diff
@@ -82,7 +82,7 @@ func validateChartName(cf *chart.Metadata) error {
 }
 
 func validateChartNameDirMatch(chartDir string, cf *chart.Metadata) error {
-	if cf.Name != filepath.Base(chartDir) {
+	if !strings.HasSuffix(chartDir, "/"+cf.Name) || cf.Name == "" {
```

Comment:
`cf.Name` would be 'nestedDir/minecraft' if the manifest for the chart was `name: nestedDir/minecraft`, which I was under the impression was desired ( or, a t least, is desired for our use ), to distinguish `fooService/backend` from `barService/backend` in the index manifest.

Re: the cf.name check, that was to prevent a false-negative, but I don't recall where.